doc/manual/src/language/derivations.md

```diff
@@ -1,161 +1,201 @@
-    `/nix/var/log/nix`.
-
-  - The builder is executed with the arguments specified by the
-    attribute `args`. If it exits with exit code 0, it is considered to
-    have succeeded.
-
-  - The temporary directory is removed (unless the `-K` option was
-    specified).
-
-  - If the build was successful, Nix scans each output path for
-    references to input paths by looking for the hash parts of the input
-    paths. Since these are potential runtime dependencies, Nix registers
-    them as dependencies of the output paths.
-
-  - After the build, Nix sets the last-modified timestamp on all files
-    in the build result to 1 (00:00:01 1/1/1970 UTC), sets the group to
-    the default group, and sets the mode of the file to 0444 or 0555
-    (i.e., read-only, with execute permission enabled if the file was
-    originally executable). Note that possible `setuid` and `setgid`
-    bits are cleared. Setuid and setgid programs are not currently
-    supported by Nix. This is because the Nix archives used in
-    deployment have no concept of ownership information, and because it
-    makes the build result dependent on the user performing the build.
+The most important built-in function is `derivation`, which is used to describe a single derivation:
+a build task to run a process on precisely defined input files and repeatably produce output files at uniquely determined file system paths.
+
+It takes as input an attribute set, the attributes of which specify the inputs to the build.
+It outputs an attribute set, and produces a [store derivation](@docroot@/glossary.md/#gloss-store-derivation) as a side effect of evaluation.
+
+<!-- FIXME: add a section on output attributes -->
+
+## Input attributes
+
+### Required
+
+- [`name`]{#attr-name} ([String](@docroot@/language/values.md#type-string))
+
+  Symbolic name for the derivation.
+  It is appended to the [store paths](@docroot@/glossary.md#gloss-store-path) of the resulting store derivation and its [output paths][output path].
+
+  Example: `name = "hello";`
+
+- [`system`]{#attr-system} ([String](@docroot@/language/values.md#type-string))
+
+  The system type on which the [`builder`](#attr-builder) executable can be run.
+
+  Nix will only build derivations where the `system` attribute matches the current [`system` configuration option].
+  It can automatically [build on other platforms](../advanced-topics/distributed-builds.md) by forwarding build requests to other machines.
+
+  Examples:
+
+  `system = "x86_64-linux";`
+
+  `system = builtins.currentSystem;`
+
+  [`builtins.currentSystem`](@docroot@/language/builtin-constants.md#builtins-currentSystem) has the value of the [`system` configuration option], and defaults to the system type of the current Nix installation.
+
+  [`system` configuration option]: @docroot@/command-ref/conf-file.md#conf-system
+
+- [`builder`]{#attr-builder} ([Path](@docroot@/language/values.md#type-path) | [String](@docroot@/language/values.md#type-string))
```

Comment:
specify that "/bin/sh" is somewhat special?

Reply:
I don't think we want people to use `/bin/sh` except through `system(2)` and such, so I'd rather push it away into a section of platform quirks.

OT: I think we should give derivation authors the option of specifying their own `/bin/sh` as a derivation attribute, at least on Linux where this is certainly feasible. Not sure about darwin.

Reply:
What's the change suggestion here?